Content.Server/_DV/CosmicCult/CosmicCultRuleSystem.cs

```diff
@@ -317,29 +320,52 @@ private void OnAntagSelect(Entity<CosmicCultRuleComponent> uid, ref AfterAntagEn
+            QueueDel(cultRule.MonumentSlowZone); // cease exist
+
             _roundEnd.EndRound(); //Woo game over yeaaaah
-            foreach (var cultist in cultRule.Cultists)
+
+            var spawnPoints = EntityManager.GetAllComponents(typeof(CosmicVoidSpawnComponent)).ToImmutableList();
+            if (spawnPoints.IsEmpty)
             {
-                if (TryComp<MobStateComponent>(cultist, out var state) && state.CurrentState != MobState.Dead)
-                {
-                    if (!TryComp<MindContainerComponent>(cultist, out var mindContainer) || !mindContainer.HasMind)
-                        return;
+                return;
+            }
 
-                    var ascendant = Spawn("MobCosmicAstralAscended", Transform(cultist).Coordinates);
-                    _mind.TransferTo(mindContainer.Mind.Value, ascendant);
-                    _metaData.SetEntityName(ascendant, Loc.GetString("cosmiccult-astral-ascendant", ("name", cultist))); //Renames cultists' ascendant forms to "[CharacterName], Ascendant"
-                    _body.GibBody(cultist); // you don't need that body anymore
-                }
+            var endQuery = EntityQueryEnumerator<HumanoidAppearanceComponent, MobStateComponent>();
+            while (endQuery.MoveNext(out var player, out _, out _))
+            {
+                var newSpawn = _rand.Pick(spawnPoints);
+                var spawnTgt = Transform(newSpawn.Uid).Coordinates;
+                Timer.Spawn(_rand.Next(TimeSpan.FromSeconds(4), TimeSpan.FromSeconds(25)), () => { EndRoundVoid(player, spawnTgt, cultRule); });
             }
-            QueueDel(cultRule.MonumentInGame); // The monument doesn't need to stick around postround! Into the bin with you.
-            QueueDel(cultRule.MonumentSlowZone); // cease exist
         }
     }
 
+    private void EndRoundVoid(EntityUid player, EntityCoordinates spawnTgt, CosmicCultRuleComponent cultRule)
+    {
+        if (!_mind.TryGetMind(player, out var mind, out _) || _mobStateSystem.IsDead(player))
+            return;
+        if (cultRule.Cultists.Contains(player))
+        {
+            var mob = Spawn(cultRule.CosmicAscended, spawnTgt);
+            _mind.TransferTo(mind, mob);
+            _metaData.SetEntityName(mob, Loc.GetString("cosmiccult-astral-ascendant", ("name", player))); //Renames cultists' ascendant forms to "[CharacterName], Ascendant"
+        }
+        else
+        {
+            var mob = Spawn(_rand.Pick(cultRule.CosmicMobs), spawnTgt);
+            _mind.TransferTo(mind, mob);
+            _metaData.SetEntityName(mob, Loc.GetString("cosmiccult-astral-minion", ("name", player))); //Renames non-cultists to "[CharacterName], Malign"
+        }
+        Spawn(cultRule.WarpVFX, spawnTgt);
+        Spawn(cultRule.WarpVFX, Transform(player).Coordinates);
+        _audio.PlayPvs(cultRule.WarpSFX, spawnTgt, AudioParams.Default.WithVolume(3f));
+        _body.GibBody(player); // you don't need that body anymore
```

Comment:
Something I haven't tested yet, how do observers work when everyone is warped to the void?
Can they warp between the two easily?

Reply:
Doesn't need changing now but worth thinking of